src/kbmod/filters.py

```diff
@@ -0,0 +1,118 @@
+from kbmod.result_list import *
+
+
+class Filter:
+    """The base class for derived filters on the ResultList."""
+
+    def get_filter_name(self):
+        """Get the name of the filter.
+
+        Returns
+        -------
+        str
+            The filter name.
+        """
+        # As a default return the class name.
+        return self.__class__
+
+    def keep_row(self, row: ResultRow):
+        """Determine whether to keep an individual row.
+
+        The derived class should overwrite this placeholder.
+
+        Parameters
+        ----------
+        row : ResultRow
+            The row to evaluate.
+
+        Returns
+        -------
+        bool
+           An indicator of whether to keep the row.
+        """
+        return True
+
```

Comment:
Sounds good. I made Filter an abstract base class.

I would like to keep the `get_filter_name` function because I think it is easier for non-python experts to understand what is going on and how it is used.